libnetwork/portallocator/portallocator.go

```diff
@@ -83,12 +83,13 @@ func Get() *PortAllocator {
 	// When this happens singleton behavior will be removed. Clients do not
 	// need to worry about this, they will not see a change in behavior.
 	once.Do(func() {
-		instance = newInstance()
+		instance = NewInstance()
 	})
 	return instance
 }
 
-func newInstance() *PortAllocator {
+// NewInstance is meant for use by libnetwork tests. It is not meant to be called directly.
+func NewInstance() *PortAllocator {
```

Comment:
I wish there was a better way to express these kind of things ("only to be used for testing"); moving to a `_test` file won't work, as those can't be called outside of a the package itself. I guess `internal/` is an option, but doesn't express "for testing only".

Marking it as `Deprecated:` to discourage use could be another one.